File: src/python/concurrent/sum.py
import os
import sys
import threading

def do_sum(path):
    _sum = 0
    with open(path, 'rb',buffering=0) as f:
        byte = f.read(1)
        while byte:
            _sum += int.from_bytes(byte, byteorder='big', signed=False)
            byte = f.read(1)
        return _sum

def thread_function(path):
    _sum = do_sum(path)
    print(path + " : " + str(_sum))

if __name__ == "__main__":
    paths = sys.argv[1:]
    threads = []

    for path in paths:
        thread = threading.Thread(target=thread_function, args=(path,))
        threads.append(thread)
        thread.start()
    
    for thread in threads:
        thread.join()

File: src/java/concurrent/Sum.java
import java.io.FileInputStream;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;

public class Sum implements Runnable {

    private String path;

    public Sum(String path) {
        this.path = path;
    }

    public static int sum(FileInputStream fis) throws IOException {
        
	    int byteRead;
        int sum = 0;
        
        while ((byteRead = fis.read()) != -1) {
        	sum += byteRead;
        }

        return sum;
    }

    public static long sum(String path) throws IOException {

        Path filePath = Paths.get(path);
        if (Files.isRegularFile(filePath)) {
       	    FileInputStream fis = new FileInputStream(filePath.toString());
            return sum(fis);
        } else {
            throw new RuntimeException("Non-regular file: " + path);
        }
    }

    @Override
    public void run() {
        long sum = 0;
        try {
            sum = sum(this.path);
        } catch (IOException e) {
            e.printStackTrace();
        }
        System.out.println(this.path + " : " + sum);
    }

    public static void main(String[] args) throws Exception {

        if (args.length < 1) {
            System.err.println("Usage: java Sum filepath1 filepath2 filepathN");
            System.exit(1);
        }

	    List<Thread> threads = new ArrayList<>();
        for (String path : args) {
            Sum sum = new Sum(path);
            Thread thread = new Thread(sum, "thread-sum");
            threads.add(thread);
            thread.start();
        }

        for (Thread thread : threads) {
            thread.join();
        }
    }
}
